code/scripts/update_default_stackage.sh

```diff
@@ -7,15 +7,17 @@ update_files() {
     local search="$3"
     local replace="$4"
 
-    find "$dir" -type f -name "$targets" | while read file; do
+    find "$dir" -type f -name "$targets" | while read -r file; do
         sed -i -E "s|$search|$replace|g" "$file"
         echo "Updated: $file"
     done
 }
 
 fetch_hlint_version() {
-    local redirect_url=$(curl -Ls -o /dev/null -w %{url_effective} "https://www.stackage.org/lts-$1/package/hlint")
-    local hlint_version=$(echo "$redirect_url" | grep -oE "hlint-[0-9]+\.[0-9]+\.[0-9]+" | sed 's/hlint-//')
+    local redirect_url
+    redirect_url=$(curl -Ls -o /dev/null -w "%{url_effective}" "https://www.stackage.org/lts-$1/package/hlint")
+    local hlint_version
+    hlint_version=$(echo "$redirect_url" | grep -oE "hlint-[0-9]+\.[0-9]+\.[0-9]+" | sed 's/hlint-//')
```

Comment:
I'm a little bit confused by these changes. What was shellcheck complaining about here?

Multi-line is less readable, and the quotation around a literal string with no special characters is unnecessary.

Reply:
It was made multi-line because of [SC2155](https://www.shellcheck.net/wiki/SC2155), which says to declare and assign separately.

The quotations were added in because of [SC1083](https://www.shellcheck.net/wiki/SC1083).

Reply:
If we wanted to handle SC2155, we can -- we would just want to add error code checking, or else the change wouldn't make any effective difference. Regarding SC1083, I see -- I can get behind that viewpoint. I write my scripts by writing out the commands myself and then copy-pasting, and editing, so any script I would write is biased towards that. Let's keep the change from SC1083, but unless we're going to change anything in regard to error checking with SC2155, we can ignore it, or, I suppose, at least slightly tweak how it's shown, i.e.,

```bash
local X
local y

x=$(...)
y=$(...)
```

Reply:
Fixed in https://github.com/JacquesCarette/Drasil/pull/3664/commits/fc3246a49571ae47c5876e7e72952adfe43335e8.

Reply:
Thanks!